barretenberg/cpp/pil/avm/alu.pil

```diff
@@ -132,7 +142,8 @@ namespace alu(256);
     a_lo * b_hi + b_lo * a_hi = partial_prod_lo + LIMB_BITS_POW * partial_prod_hi;
 
     // This holds the product over the integers
-    pol PRODUCT = a_lo * b_lo + LIMB_BITS_POW * partial_prod_lo + MAX_BITS_POW * (partial_prod_hi + a_hi * b_hi);
+    // (u1 multiplication only cares about a_lo and b_lo)
+    pol PRODUCT = a_lo * b_lo + (1 - u1_tag) * (LIMB_BITS_POW * partial_prod_lo + MAX_BITS_POW * (partial_prod_hi + a_hi * b_hi));
```

Comment:
We should not need the exception for u1, ie., the term (1 - u1_tag) can be removed I think.

Reply:
Without that exception for u1, I believe `a_hi` and `b_hi` are underconstrained since `LIMB_BITS_POW ` is 0. Or am I wrong?